version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: anivalientes-web
    volumes:
      - .:/var/www
    networks:
      - anivalientes
    depends_on:
      - mysql

  web:
    image: nginx:stable
    container_name: anivalientes-nginx
    ports:
      - "8080:80"
    volumes:
      - .:/var/www
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - app
    networks:
      - anivalientes

  mysql:
    image: mysql:8
    container_name: anivalientes-mysql
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: dr1cHT@van
      MYSQL_DATABASE: anivalientes_db
      MYSQL_USER: anivalientes
      MYSQL_PASSWORD: dr1cHT@van
    ports:
      - "3307:3306"  # Cambiado para evitar conflicto con MySQL local
    volumes:
      - mysql-data:/var/lib/mysql
    networks:
      - anivalientes

  phpmyadmin:
    image: phpmyadmin/phpmyadmin
    container_name: anivalientes-pma
    restart: unless-stopped
    ports:
      - "8081:80"
    environment:
      PMA_HOST: mysql
      MYSQL_ROOT_PASSWORD: dr1cHT@van
    depends_on:
      - mysql
    networks:
      - anivalientes

networks:
  anivalientes:
    driver: bridge

volumes:
  mysql-data:
